query/src/main/java/org/infinispan/query/affinity/AffinityShardIdentifierProvider.java

```diff
@@ -121,6 +122,9 @@ private DistributionManager getDistributionManager() {
 
    private static Integer getNumberOfShards(Properties properties) {
       String nShards = properties.getProperty(NUMBER_OF_SHARDS_PROP);
+      if (nShards == null) {
+         nShards = properties.getProperty(SHARDING_STRATEGY_PROP + "." + NUMBER_OF_SHARDS_PROP);
```

Comment:
By passing the default number of shard as a 2nd arg to getProperty we can ensure getNumberOfShards always has a non-null value to return, so it can return int